packages/transformers/babel/src/config.js

```diff
@@ -1,141 +1,191 @@
-import getJSXConfig from './jsx';
-import getFlowConfig from './flow';
+import nullthrows from 'nullthrows';
 import path from 'path';
+import {loadPartialConfig, createConfigItem} from '@babel/core';
+
+import {IConfig} from '@parcel/types';
+import {md5FromObject} from '@parcel/utils';
+
+import getEnvOptions from './env';
+import getJSXOptions from './jsx';
+import getFlowOptions from './flow';
+import getTypescriptOptions from './typescript';
+import type {BabelConfig} from './types';
 
 const TYPESCRIPT_EXTNAME_RE = /^\.tsx?/;
-const NODE_MODULES = `${path.sep}node_modules${path.sep}`;
-
-export default async function getBabelConfig(asset: MutableAsset) {
-  // Consider the module source code rather than precompiled if the resolver
-  // used the `source` field, or it is not in node_modules.
-  let pkg = await asset.getPackage();
-  let isSource =
-    !!(
-      pkg &&
-      pkg.source &&
-      (await asset.fs.realpath(asset.filePath)) !== asset.filePath
-    ) || !asset.filePath.includes(NODE_MODULES);
-
-  // Try to resolve a .babelrc file. If one is found, consider the module source code.
-  let babelrc = await getBabelRc(asset, pkg, isSource);
-  isSource = isSource || !!babelrc;
-
-  let result = {};
-  mergeConfigs(result, babelrc);
-
-  // Typescript must use the plugin directly (not the typescript preset) and must
-  // come before preset env, otherwise proposals and nonstandard syntax is not
-  // transformed in time.
-  if (path.extname(asset.filePath).match(TYPESCRIPT_EXTNAME_RE)) {
-    let hasTypescript =
-      babelrc &&
-      (hasPlugin(babelrc.config.presets, [
-        '@babel/typescript',
-        '@babel/preset-typescript'
-      ]) ||
-        hasPlugin(babelrc.config.plugins, [
-          '@babel/transform-typescript',
-          '@babel/plugin-transform-typescript'
-        ]));
-
-    if (!hasTypescript) {
-      mergeConfigs(result, {
-        internal: true,
-        babelVersion: 7,
-        config: {
-          plugins: [
-            [
-              require('@babel/plugin-transform-typescript'),
-              {isTSX: path.extname(asset.filePath) === '.tsx'}
-            ]
-          ]
-        }
+const BABEL_TRANSFORMER_DIR = path.dirname(__dirname);
+
+export async function load(config: IConfig) {
+  let partialConfig = loadPartialConfig({filename: config.searchPath});
+  if (partialConfig && partialConfig.hasFilesystemConfig()) {
+    let {babelrc, config: configjs} = partialConfig;
+    let {canBeRehydrated, dependsOnRelative, dependsOnLocal} = getStats(
+      partialConfig.options
+    );
+
+    // babel.config.js files get required by @babel/core so there's no use in including it for watch mode invalidation
+    if (babelrc != null && configjs == null) {
+      config.setResolvedPath(babelrc);
+    } else if (configjs) {
+      // TODO: warn and invalidate on startup
+    }
+
+    if (babelrc && (await isExtended(/*babelrc*/))) {
+      // TODO: warn and invalidate on startup
```

Comment:
why do we need to invalidate if you used `extends`? Would we not just add all the extended files as included files?

Reply:
I guess I was thinking that there could be many chained extensions and I didn't want to keep following through to all of them when `@babel/core` already did that. I'd rather `@babel/core` just tell us. But yeah, might be something worth doing.

Reply:
Oh yeah for sure. We should make a PR to babel for that.